ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -291,14 +291,13 @@ public void processPriorCommands(final PersistentQueryCleanupImpl queryCleanup)
           .getKsqlEngine()
           .getPersistentQueries();
 
-      queryCleanup.cleanupLeakedQueries(queries);
-
       if (commandStore.corruptionDetected()) {
         LOG.info("Corruption detected, queries will not be started.");
         queries.forEach(QueryMetadata::setCorruptionQueryError);
       } else {
         LOG.info("Restarting {} queries.", queries.size());
         queries.forEach(PersistentQueryMetadata::start);
+        queryCleanup.cleanupLeakedQueries(queries);
```

Comment:
can you also leave a comment explaining why it's so important that we only do this here